### PR TITLE
Revert changes to STS leases but keep the ttl field

### LINKED ISSUE #19513: AWS secrets engine no longer returns lease_duration for STS credentials
**Describe the bug**

https://github.com/hashicorp/vault/pull/15869 changed the `aws` secrets engine such that it no longer returns leases for STS credentials, which makes sense. However, it also no longer returns a `lease_duration`.

**To Reproduce**

```sh
$ vault server -dev
```

```sh
$ vault secrets enable aws

$ vault write aws/config/root access_key=<redacted> secret_key=<redacted> region=us-east-1

$ vault write aws/roles/my-role credential_type=assumed_role role_arns="<redacted>" max_sts_ttl=900

$ vault read aws/sts/my-role
```

**Expected behavior**
A `lease_duration` is returned with the STS credentials.

**Environment:**
* Vault Server Version (retrieve with `vault status`): `Vault v1.13.0 (a4cf0dc4437de35fce4860857b64569d092a9b5a), built 2023-03-01T14:58:13Z`
* Vault CLI Version (retrieve with `vault version`): `Vault v1.13.0 (a4cf0dc4437de35fce4860857b64569d092a9b5a), built 2023-03-01T14:58:13Z`
* Server Operating System/Architecture: Ubuntu 22.04.2 LTS on AMD64

Vault server configuration file(s):

N/A

**Additional context**

The KV v1 engine returns a `lease_duration` without a `lease_id`, so there is precedent for this.

### PR DESCRIPTION
**Pull request by kschoche**
* Reverts lease changes brought in by #15869 while keeping the `ttl` changes.
* Fixes #19513

### CODE REVIEW

File: changelog/20034.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note: bug
+secrets/aws: Revert changes that removed the lease on STS credentials, while leaving the new ttl field in place. [[GH-20034](https://github.com/hashicorp/vault/pull/20034)]
```
Comment (calvn) [inline changelog/20034.txt:2]: I don't think the link is needed. The CL generation will take care of that.

```suggestion
secrets/aws: Revert changes that removed the lease on STS credentials, while leaving the new ttl field in place.
```

Comment (kschoche) [reply changelog/20034.txt:2]: Thanks!

**Review by austingebauer — APPROVED** [review]
LGTM

**Comment by alex-ikse** [comment]
Will fix: https://github.com/hashicorp/terraform-provider-vault/issues/1805

**Review by robmonte — APPROVED** [review]
👍

**Comment by alex-ikse** [comment]
> Wouldn't it be better to aim for a solution that gave clients the necessary compatible feedback in responses, but **also** retained the optimization?

Terraform Vault provider must be updated before doing this kind of change. See: https://github.com/hashicorp/terraform-provider-vault/issues/1805

**Pull request merged**